### PR TITLE
made some of the math optional + aded max output

### PR DESCRIPTION
**Pull request by FelixMo42**
- made accumulator math optional if not using kI
- let user set max output size

### CODE REVIEW

File: src/main/kotlin/org/team5419/fault/math/pid/PIDF.kt [hunk 1]
```diff
@@ -59,16 +69,45 @@ public class PIDF {
...
         }
+    }
+
+    public fun calculate(): Double {
+        var total = 0.0
+
+        if (kP != 0.0) {
+            total += kP * error
+        }
+
```
Comment (andycate) [inline src/main/kotlin/org/team5419/fault/math/pid/PIDF.kt:80]: Why add the if statements?

**Review by andycate — CHANGES_REQUESTED** [review]
Looks good. Just one question.

File: src/main/kotlin/org/team5419/fault/math/pid/PIDF.kt (hunk 1 above)
Comment (FelixMo42) [reply src/main/kotlin/org/team5419/fault/math/pid/PIDF.kt:80]: Admittedly, that if statement is basically useless. It's just their for consistency

File: src/main/kotlin/org/team5419/fault/math/pid/PIDF.kt [hunk 2]
```diff
@@ -8,6 +8,7 @@ public class PIDF {
     public var kI: Double
     public var kD: Double
     public var kF: Double
+    public var totalMax: Double
```
Comment (BBScholar) [inline src/main/kotlin/org/team5419/fault/math/pid/PIDF.kt:11]: I personally dont thing the PIDF controller itself should cap the output.

Comment (BBScholar) [reply src/main/kotlin/org/team5419/fault/math/pid/PIDF.kt:11]: More elaboration: The subsystem itself should be the one that caps output. The subsystem should be the only class that knows the physical parameters of itself, adding an output cap to the pidf controller effectively defines the max output voltage in an auxiliary class.

**Review by BBScholar — CHANGES_REQUESTED** [review]
Looks fine. The thing about pid vs subsystem capping is just my opinion. Ask Andy

File: src/main/kotlin/org/team5419/fault/math/pid/PIDF.kt (hunk 2 above)
Comment (FelixMo42) [reply src/main/kotlin/org/team5419/fault/math/pid/PIDF.kt:11]: Isn't the point of this class to increase readability while reducing the need to rewrite code?
I think allowing people to get the PDF do it while making neater and reduce the risk of bugs by having people misemployment it in the future.

Comment (andycate) [reply src/main/kotlin/org/team5419/fault/math/pid/PIDF.kt:11]: I think it’s good for now. Let’s merge

**Pull request merged**